.github/workflows/image-publish.yml

```diff
@@ -42,6 +43,15 @@ jobs:
     needs: build
     runs-on: ubuntu-20.04
     steps:
+      - name: Set variables:
+        run: |
+          BRANCH_NAME=${{ github.ref }}
+          BRANCH_NAME=${BRANCH_NAME##*/}
+          if [[ $BRANCH_NAME == 'refs/heads/main' ]]; then
```

Comment:
If you're setting BRANCH_NAME to only equal `main` instead of `refs/heads/main` then your if-check on line 50 is going to fail because you removed `refs/heads/` on line 49